msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1064,51 +1060,178 @@ protected void acquireTokenSilent(@NonNull final String[] scopes,
+
+            if (isHomeTenantEquivalent(tenantIdNameOrAlias)
+                    || isAccountHomeTenant(multiTenantAccount.getClaims(), tenantIdNameOrAlias)) {
+                // Use home...
+                validateClaimsExistForTenant(tenantIdNameOrAlias, multiTenantAccount);
+
+                accountRecord.setLocalAccountId(multiTenantAccount.getId());
+                accountRecord.setUsername(
+                        SchemaUtil.getDisplayableId(multiTenantAccount.getClaims())
+                );
+
+                return accountRecord;
+            } else if (null != multiTenantAccount.getTenantProfiles().get(tenantIdNameOrAlias)) {
+                // Use this guest...
+                final ITenantProfile profileForRequest =
+                        multiTenantAccount
+                                .getTenantProfiles()
+                                .get(tenantIdNameOrAlias);
+
+                validateClaimsExistForTenant(tenantIdNameOrAlias, profileForRequest);
+
+                accountRecord.setLocalAccountId(profileForRequest.getId());
+                accountRecord.setUsername(
+                        SchemaUtil.getDisplayableId(profileForRequest.getClaims())
+                );
+
+                return accountRecord;
+            } else {
+                // We need to query metadata...
+                final OpenIdProviderConfigurationClient client =
+                        new OpenIdProviderConfigurationClient(requestAuthority);
+
+                final OpenIdProviderConfiguration providerConfiguration =
+                        client.loadOpenIdProviderConfiguration();
+
+                final String issuer = providerConfiguration.getIssuer();
+                final Uri issuerUri = Uri.parse(issuer);
+                final List<String> paths = issuerUri.getPathSegments();
+                final String tenantPath = paths.get(0);
+
+                if (multiTenantAccount.getTenantId().equals(tenantPath)) {
+                    // use home...
+                    validateClaimsExistForTenant(tenantPath, multiTenantAccount);
+
+                    accountRecord.setLocalAccountId(multiTenantAccount.getId());
+                    accountRecord.setUsername(
+                            SchemaUtil.getDisplayableId(multiTenantAccount.getClaims())
+                    );
+                } else {
+                    // Use the matching profile
+                    final ITenantProfile profileForRequest =
+                            multiTenantAccount
+                                    .getTenantProfiles()
+                                    .get(tenantPath);
+
+                    validateClaimsExistForTenant(tenantPath, profileForRequest);
+
+                    accountRecord.setLocalAccountId(profileForRequest.getId());
+                    accountRecord.setUsername(
+                            SchemaUtil.getDisplayableId(profileForRequest.getClaims())
```

Comment:
same here

Reply:
Resolved in f9d215149496fc299f2f606a73aa81e82f51fe73